### PR TITLE
Auto Balence Options

### PR DESCRIPTION
**Pull request by Sahiltheram**

### CODE REVIEW

File: networktables.json [hunk 1]
```diff
@@ -0,0 +1 @@
+[]
```
Comment (batchen1) [inline networktables.json]: delete this file.
When you add+commit+push, don't just do it automatically and add everything you changed. You have to look and select the files that you changed, then look at what you changed to confirm it is good and only then you add and commit them. These files are related to simulation and we don't want them in our branch.

File: simgui-ds.json [hunk 2]
```diff
@@ -0,0 +1,92 @@
+{
```
Comment (batchen1) [inline simgui-ds.json]: same

File: simgui.json [hunk 3]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (batchen1) [inline simgui.json]: same

File: src/main/java/frc/robot/AutonomousChooser.java [hunk 4]
```diff
@@ -63,12 +63,142 @@ public void addOptions() {
...
         locationChooser.addOption(Location.Right.name(), Location.Right);
     }
+
+    public void setOdometry(Drivetrain drivetrain, Location location, Action action, Alliance alliance) {
+        double x = 0;
+        double y = 0;
+        if (alliance == Alliance.Blue) {
+            
+            if(action == Action.CrossLine && location == Location.Left) {
+                x = 72;
```
Comment (batchen1) [inline src/main/java/frc/robot/AutonomousChooser.java]: we are always setting x to 72, why not just put it outside the if?

File: src/main/java/frc/robot/AutonomousChooser.java [hunk 5]
```diff
@@ -63,12 +63,142 @@ public void addOptions() {
...
+
+            else {
+                x = 72;
+                y = 108;
+            }
+        }
+        
+        
+        
+        else if (alliance == Alliance.Red) {
```
Comment (batchen1) [inline src/main/java/frc/robot/AutonomousChooser.java]: for red, we just need to add 99 to each option, no need to duplicate all the logic.

File: src/main/java/frc/robot/AutonomousChooser.java [hunk 6]
```diff
@@ -114,6 +244,15 @@ else if (action == Action.OnePieceMoveLeft) {
         else if (action == Action.OnePieceMoveRight) {
             return new OneGamepiece(drivetrain, arm, extender, gripper, -1, location, allianceColor);
         }
+        else if (action == Action.DepositOneAndBalanceRight) {
```
Comment (batchen1) [inline src/main/java/frc/robot/AutonomousChooser.java]: now that we set odometry in the autonomous choose, our autonomous commands no longer need the alliance color as a parameter. Can you refactor and delete it?

File: src/main/java/frc/robot/commands/Autonomous/Balance.java [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+import frc.robot.subsystems.Extender;
+import frc.robot.subsystems.GripperSubsystem;
+
+public class Balance extends SequentialCommandGroup {
+    
+    public Balance (Drivetrain drivetrain, Arm arm, Extender extender, GripperSubsystem gripper, AutonomousChooser.Location location, Alliance allianceColor) {
+
+        addCommands(
+            new ResetEncoders(arm, extender),
+            new MoveDistanceSpinTraj(drivetrain, 0, 3, Math.toRadians(180)),
```
Comment (batchen1) [inline src/main/java/frc/robot/commands/Autonomous/Balance.java]: I don't think we need to trajectory here, we assume drive team positioned the robot in the middle of the platform and that we can just start the balance command.

File: src/main/java/frc/robot/commands/Autonomous/DepositOneAndBalance.java [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+            new ParallelRaceGroup(
+                new SequentialCommandGroup(
+                    new ExtendToPosition(extender, ArmPositionGrid.TOP_MIDDLE.getExtenderPosition()),
+                    new OpenGripper(gripper)
+                ),
+                new HoldArmPID(arm, ArmPositionGrid.TOP_MIDDLE.getArmPosition())
+            ),
+
+            new Stow(arm, gripper, extender),
+            new MoveDistanceSpinTraj(drivetrain, 0.2 * direction, 1, Math.toRadians(180)),
```
Comment (batchen1) [inline src/main/java/frc/robot/commands/Autonomous/DepositOneAndBalance.java]: we want to move 10cm forward (+x) and 20 cm on the y depending on direction (positive for moving right and negative for moving left), so the MoveDistanceSpinTraj should get 0.1 for the x and 0.2*direction for y.

**Review by batchen1 — CHANGES_REQUESTED** [review]

**Review by batchen1 — APPROVED** [review]

**Review by bengold22 — APPROVED** [review]

**Pull request merged**